src/Resources/app/administration/src/module/sw-in-app-purchases/component/sw-in-app-purchase-checkout/index.ts

```diff
@@ -0,0 +1,164 @@
+    errors: Array<{
+        status: string;
+        code: string;
+        title: string;
+        detail: string;
+        meta: {
+            parameters: string[];
+        };
+        trace: Array<{
+            file: string;
+            line: number;
+            function: string;
+            class: string;
+            type: string;
+            args?: string[];
+        }>;
+    }>;
+}
+
+export default Shopware.Component.wrapComponentConfig({
+    template,
+
+    inject: [
+        'inAppPurchasesService'
+    ],
+
+    mixins: [
+        Shopware.Mixin.getByName('notification')
+    ],
+
+    data() {
+        return {
+            state: 'loading' as 'loading' | 'purchase' | 'error' | 'success',
+            store: Shopware.Store.get('inAppPurchaseCheckout'),
+            inAppPurchaseCart: null as IAP.InAppPurchaseCart | null,
+            extension: null as IAP.Extension | null,
+            tosAccepted: false,
+            errorSnippet: null as string | null
+        };
+    },
+
+    created() {
+        this.createdComponent();
+    },
+
+    computed: {
+        priceModel(): IAP.InAppPurchasePriceModel | null {
+            return this.inAppPurchaseCart?.positions?.[0].feature.priceModel || null;
+        },
+        purchase(): IAP.InAppPurchase | null {
+            return this.inAppPurchaseCart?.positions?.[0].feature || null;
+        },
+        assetFilter() {
+            return Shopware.Filter.getByName('asset');
+        },
+        extensionIcon() {
+            if (this.extension?.icon) {
+                return this.extension.icon;
+            }
+
+            if (this.extension?.iconRaw) {
+                return `data:image/png;base64, ${this.extension.iconRaw}`;
+            }
+
+            return this.assetFilter('administration/static/img/theme/default_theme_preview.jpg');
+        }
+    },
+
+    methods: {
+        createdComponent() {
+            this.store.$subscribe(() => {
+                this.requestFeature();
+            });
+        },
+
+        async requestFeature() {
+            if (!this.store.extension || !this.store.entry) {
+                this.reset();
+
+                return;
+            }
+
+            this.state = 'loading';
+
+            await Promise.all([
+                this.cart = this.inAppPurchasesService.createCart(
+                    this.store.extension.name,
+                    this.store.entry.identifier
+                ),
+                this.inAppPurchasesService.getExtension(this.store.extension.name)
+            ]).then(([inAppPurchaseCart, extension]) => {
+                this.inAppPurchaseCart = inAppPurchaseCart;
+                this.extension = extension;
+                this.state = 'purchase';
+            }).catch((errorResponse: ErrorResponse) => {
+                Shopware.Utils.debug.error('InAppPurchases', errorResponse);
+                this.state = 'error';
+                this.errorSnippet = errorResponse?.errors[0]?.detail;
```

Comment:
not sure if we want this, it's untranslated and very verbose :thinking: maybe logging is enough then

Reply:
logging should be enough in my opinion